### PR TITLE
Remove Scala 2.9 and add Scala 2.11 version

### PR DESCRIPTION
**Pull request by cramertj**

### CODE REVIEW

File: project/Dependencies.scala [hunk 1]
```diff
@@ -5,10 +5,9 @@ object Dependencies {
     val asyncHttpClient = "1.7.5"
     val jodaConvert = "1.2"
     val jodaTime = "2.1"
-    val rojomaJson_29 = "2.0.0"
-    val rojomaJson_210 = "[2.1.0,3.0.0)"
+    val rojomaJson = "[2.1.0,3.0.0)"
     val scalaCheck = "1.10.0"
-    val scalaTest = "1.9.1"
+    val scalaTest = "2.2.6"
```
Comment (cramertj) [inline project/Dependencies.scala]: Had to bump this to get a 2.11 compatible version

File: project/ScalaVersion.scala [hunk 2]
```diff
@@ -1,10 +1,10 @@
 sealed abstract class MajorScalaVersion
-case object Scala29 extends MajorScalaVersion
 case object Scala210 extends MajorScalaVersion
+case object Scala211 extends MajorScalaVersion
 
 object ScalaVersion {
   def v(implicit sv: String) =
-    if(sv startsWith "2.9.") Scala29
-    else if(sv startsWith "2.10.") Scala210
+    if(sv startsWith "2.10.") Scala210
```
Comment (cramertj) [inline project/ScalaVersion.scala]: This is unnecessary now that I've removed the casing on rojoma-json version. Mind if I remove?

Comment (rjmac) [inline project/ScalaVersion.scala]: Go ahead; it was just a hack due to a change in the way binaries were marked in 2.9 vs 2.10 IIRC.

File: soda-consumer-scala/src/main/scala/com/socrata/soda2/consumer/http/HttpConsumer.scala [hunk 3]
```diff
@@ -10,8 +10,6 @@ import com.socrata.future.ExecutionContextTimer
 
 /** An implementation of [[com.socrata.soda2.consumer.Consumer]] which operates on a real HTTP server. */
 class HttpConsumer(lowLevel: LowLevelHttp) extends Consumer(lowLevel) {
-  private[socrata] def this(client: AsyncHttpClient, logicalHost: String, physicalHost: String, port: Int = 443, secure: Boolean = true, authorization: Authorization = NoAuth)(implicit executionContext: ExecutionContext, timer: ExecutionContextTimer) =
```
Comment (cramertj) [inline soda-consumer-scala/src/main/scala/com/socrata/soda2/consumer/http/HttpConsumer.scala]: Both this and the private constructor for HttpPublisher were error-ing with " multiple overloaded alternatives of constructor HttpConsumer define default arguments." This [isn't allowed](https://stackoverflow.com/questions/4652095/why-does-the-scala-compiler-disallow-overloaded-methods-with-default-arguments).

File: project/BuildSettings.scala [hunk 4]
```diff
@@ -9,7 +9,7 @@ object BuildSettings {
     Seq(
       version := "2.0.0-SNAPSHOT",
       scalaVersion := "2.10.1",
```
Comment (rjmac) [inline project/BuildSettings.scala]: This should probably be changed to 2.10.4

File: project/BuildSettings.scala [hunk 5]
```diff
@@ -9,7 +9,7 @@ object BuildSettings {
...
-      crossScalaVersions := Seq("2.9.3", "2.10.1"),
+      crossScalaVersions := Seq("2.10.4", "2.11.8"),
```
Comment (rjmac) [inline project/BuildSettings.scala]: ..and if you want to dedup, this can be `crossScalaVersions := Seq(scalaVersion.value, "2.11.8")` I believe.

Comment (cramertj) [inline project/BuildSettings.scala]: Sure.

**Comment by rjmac** [comment]
:boat:

**Pull request merged**